### DISCUSSION TITLE
Instalar programas para Fundamentos SQL Server.

### DISCUSSION THREAD

**Discussion by S-Dornelas** [balta-io/forum#179] [Tira Dúvidas] [answered]
Não consigo instalar os programas para praticar os exercícios e aulas do curso de SQL Server, quando instalo o wsl (Ubuntu) apresenta a seguinte mensagem: _O Subsistema do Windows para Linux não tem distribuições instaladas. As distribuições podem ser instaladas visitando a Microsoft Store: https://aka.ms/wslstoree_ 

E quando instalo o Docker apresenta a seguinte mensagem: _Hardware assisted virtualization and data execution protection must be enabled in the BIOS. See https://docs.docker.com/desktop/windows/troubleshoot/#virtualization_

Segui todos os paços que o artigo diz pra instalar os programas e não pega de jeito algum.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @S-Dornelas, como vai?

Se você estiver nas últimas versões do Windows 11, pode executar o seguinte comando (Como Administrador):
```
wsl --install
```

Se já estiver com WSL instalado e quiser instalar uma distro como o Ubuntu. Conforme o erro te sugere, basta acessar a loja do Windows e buscar por Ubuntu.

Em relação ao Docker, você precisa habilitar a virtualização na BIOS, este processo muda de acordo com cada fabricante de processador/placa mãe. Qual configuração você usa?

**Reply by S-Dornelas** [reply]
Eu já instalei o wsl 2, bem como baixei o Ubuntu no app store, mas não pega, apresenta a seguinte mensagem: 
Installing, this may take a few minutes... 
WslRegisterDistribution failed with error: 0x800701bc
Error: 0x800701bc WSL 2 requer uma atualizaçãoo para seu componente kernel. Para obter mais informações, visite https://aka.ms/wsl2kernel

Fui no tutorial que o site indica e fiz as devidas alterações de requisitos para execução do wsl 2, e nada. Até atualizei o windows para o 11 pra ver se ajuda, já que para aceitar o wsl 2 tem uma versão do windows 10 que dar problemas.

A configuração do meu PC é: Placa mãe A520M DS3H - GIGABYTE (atualizada com a Bios F16a, PROCESSADOR AMD RYZEM 5 5600G.

**Comment by andrebaltieri** [comment]
Boa tarde, @S-Dornelas, como vai?

Certo, você baixou e instalou a atualização conforme consta aqui:
https://aka.ms/wsl2kernel
Precisa reiniciar o PC depois.

Sobre habilitar a virtualização, nos AMD é chamado SVM. Aqui tem um tutorial:
https://www.goodtechmaster.com/how-to-enable-virtualization-amd-v-on-ryzen-processors/

Lembrando que se estiver instalando o Docker/WSL apenas por conta do SQL Server, não precisa. Tem uma versão para Windows do Express:
https://www.microsoft.com/pt-br/download/details.aspx?id=101064

**Comment by S-Dornelas** [comment]
Prezados, boa tarde.

Só consegui fazendo a alteração da BIOS mesmo, muito obrigado pela atenção.

*Cordialmente.*

*Sandro José Vitorino Dornelas*
*Fone: (81) 99433-2276*
*E-mail: ***@***.*** ***@***.***>*


Em ter., 20 de dez. de 2022 às 14:02, Andre Baltieri <
***@***.***> escreveu:

**Reply by andrebaltieri** [reply]
Boa tarde, @S-Dornelas, perfeito. Habilitar o SVM e um passo obrigatório para instalação do Docker.
